### ISSUE TITLE
[VL] GHA is not stable when downloading Spark package

### ISSUE THREAD

**Issue by zhouyuan** [apache/incubator-gluten#8266]
### Backend

VL (Velox)

### Bug description

```
bash .github/workflows/util/install_spark_resources.sh 3.5
  shell: sh -e {0}
  env:
    ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true
    MVN_CMD: mvn -ntp
    WGET_CMD: wget -nv
    SETUP: bash .github/workflows/util/setup_helper.sh
    CCACHE_DIR: /home/runner/work/incubator-gluten/incubator-gluten/.ccache
failed: Connection timed out.
failed: Network is unreachable.
Error: Process completed with exit code [4](https://github.com/apache/incubator-gluten/actions/runs/12389480827/job/34583002711?pr=8265#step:6:4).
```

### Spark version

None

**Comment by zhouyuan** [comment]
archive.apache.org is not stable
https://github.com/apache/incubator-gluten/blob/main/.github/workflows/util/install_spark_resources.sh#L27

**Comment by zjuwangg** [comment]
I also just opened a discussion https://github.com/apache/incubator-gluten/discussions/8275. Glad to see the issue and solution!

**Comment by zhouyuan** [comment]
Notes on several tests:
- retry on wget doesn't help
- the issue is due to we launch these jobs concurrently, so apache.org banned some IP
- Hosting the package to other place or move into docker can fix it

**Comment by PHILO-HE** [comment]
I will try to pre-install these resources into docker image today. Assume the docker size limit will not be hit. cc @zjuwangg

**Comment by zhouyuan** [comment]
fixed by #8294
